src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -47,7 +48,17 @@ public WebClientDownloader(string userName, string password, ILogger logger)
 
             if (this.client == null)
             {
-                this.client = new HttpClient();
+                if (clientCertPath != null || clientCertPassword != null) // password mandatory, as to use client cert in .jar it cannot be with empty password
```

Comment:
So maybe && instead ? Otherwise raise an error.

Reply:
@mickael-caro-sonarsource, I think we cannot raise an error here, case the params are not there, it will use a new object anyway.

Reply:
Ok, but still i think cert path cannot come without password as you said, so better to check if both are not null.

Reply:
I'd also use `&&`. If somebody sets only password, the code will try to create certificate from `null` path.

Reply:
Ah ok, makes sense, since the .jar cannot accept certs without pw, so added in the condition &&